eZ/Publish/API/Repository/Tests/FieldType/RichTextIntegrationTest.php

```diff
@@ -618,6 +620,103 @@ public function testConvertRemoteObjectIdToObjectId($test, $expected)
         );
     }
 
+    /**
+     * @param string $xmlDocumentPath
+     * @dataProvider providerForTestCreateContentWithValidCustomTag
+     */
+    public function testCreateContentWithValidCustomTag($xmlDocumentPath)
+    {
+        $validXmlDocument = $this->createDocument($xmlDocumentPath);
+        $this->createContent(new RichTextValue($validXmlDocument));
+    }
+
+    /**
+     * Data provider for testCreateContentWithValidCustomTag.
+     *
+     * @return array
+     */
+    public function providerForTestCreateContentWithValidCustomTag()
+    {
+        $data = [];
+        $iterator = new \DirectoryIterator(__DIR__ . '/_fixtures/ezrichtext/custom_tags/valid');
+        foreach ($iterator as $fileInfo) {
+            if ($fileInfo->isFile() && $fileInfo->getExtension() === 'xml') {
+                $data[] = [
+                    $fileInfo->getRealPath(),
+                ];
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param string $xmlDocumentPath
+     * @param string $expectedValidationMessage
+     *
+     * @dataProvider providerForTestCreateContentWithInvalidCustomTag
+     */
+    public function testCreateContentWithInvalidCustomTag(
+        $xmlDocumentPath,
+        $expectedValidationMessage
+    ) {
+        $this->expectException(\RuntimeException::class);
```

Comment:
Can't this be just tested with assert for message in catch block? This seems a little bit weird for me.

Reply:
@ViniTou This is a pretty standard way of asserting exceptions occurrence in PHPUnit. I don't think using `try {} catch()` in a test case is a good idea.

Reply:
@webhdx 
yeah, I know. I was asking about checking for exception which is thrown in that same function.

Reply:
> Can't this be just tested with assert for message in catch block? This seems a little bit weird for me.

Hmm, maybe... Here, it's a little bit artificial because `ValidationError` is not throwable. 
For this to be "clean" I would need spaghetti code that creates actual expected exception instance (`ContentFieldValidationException` with nested `ValidationError`), which contradicts the concept of clean code ;)

On the other hand, asserting message would also mean adding extra `fail` line at the end of the function and return in `catch` block, thus more code.

Moreover `expectException` at the top of a test method indicates the purpose of that method.

I'm a little bit unsure about it myself. Any POV from the Other Reviewers? :)

Reply:
Well.. How would the try catch look like? what would you type hint? expectException only takes class names and not interfaces /abstract, or?

Reply:
> Well.. How would the try catch look like? what would you type hint? expectException only takes class names and not interfaces /abstract, or?

It can hint abstracts and interfaces. My issue here is that something which tests a failure should test an occurence of an exception. That exception occurs, but the expected message is nested within it and difficult to model as expected value. Moreover it's an instance of `ValidationError` which is not `Throwable`.

@andrerom @ViniTou  I've created a fixup with an alternative solution 2c3982d. WDYT?

Reply:
This is what I had in my mind at first, dunno if it is more clear but i think it is more readable and straightforward that way.

Reply:
I also think this is a tad more readable, so +1 form my side, even if further refactoring might make it even more clear.